libs/domain/feature-loaders/src/snapshot/SnapshotProposalLoader.ts

```diff
@@ -0,0 +1,187 @@
+import { createGraphQLClient, GraphQLClient, ProgramError } from "@banklessdao/util-data";
+import { InitContext, InternalLoadContext, LoadContext, ScheduleMode } from "@shared/util-loaders";
+import { Data, Nested, NonEmptyProperty, OptionalNumberArrayOf, OptionalObjectRef, OptionalProperty, OptionalStringArrayOf, Property, RequiredArrayRef, RequiredStringArrayOf } from "@banklessdao/util-schema";
+import { DocumentNode } from "graphql";
+import gql from "graphql-tag";
+import * as t from "io-ts";
+import * as TE from "fp-ts/TaskEither";
+import { GraphQLDataLoaderBase } from "../base/GraphQLDataLoaderBase";
+import { BATCH_SIZE } from "../defaults";
+import { withMessage } from "io-ts-types";
+
+export const URL = "https://hub.snapshot.org/graphql";
+
+export const makeQUERY: (space:String)=>DocumentNode = (space) => {
+    space = `"${space}"` // So it gets parsed correctly
+    return gql`
+        query snapshotProposals($limit: Int,$cursor: Int) {
+            proposals (
+                first: $limit,
+                where: {
+                    space_in: [${space}],
+                },
+                skip:$cursor,
+                orderBy: "created",
+                orderDirection: asc
+            ) {
+                id
+                strategies {
+                    name
+                    params
+                }
+                title
+                body
+                choices
+                created
+                start
+                end
+                snapshot
+                state
+                author
+                space {
+                    id
+                    name
+                }
+            }
+    }
+    `;
+}
+
+// Helper for optional fields
+const optional = <T extends t.Mixed>(x:T)=>t.union([x,t.undefined])
```

Comment:
Now this is very smart 👍  I think we can also extract this to one of the `util` packages where we keep the library code.

Reply:
Moved this to library